draft-ietf-rtcweb-jsep.xml

```diff
@@ -2557,31 +2557,125 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
 
           <t>The first step in generating an initial answer is to
           generate session-level attributes. The process here is
-          identical to that indicated in the Initial Offers section
+          identical to that indicated in the initial offers section
           above, except that the "a=ice-options" line, with the
           "trickle" option as specified in
           <xref target="I-D.ietf-ice-trickle"></xref>, Section 4, is
           only included if such an option was present in the offer.</t>
 
-          <t>The next step is to generate session-level lip sync groups
-          as defined in
-          <xref target="RFC5888" />, Section 7. For each group of type
-          "LS" present in the offer, determine which of the local
-          RtpTransceivers identified by that group's mid values
-          reference a common local MediaStream (as specified in the
-          addTrack and addTransceiver methods). If at least two such
-          RtpTransceivers exist, a group of type "LS" with the mid
-          values of these RtpTransceivers MUST be added. Otherwise,
-          this indicates a difference of opinion between the offerer
-          and answerer regarding lip sync status, and as such, the
-          offered group MUST be ignored and no corresponding "LS" group
-          generated.</t>
+          <t>The next step is to generate session-level lip sync
+          groups, as defined in <xref target="RFC5888" />, Section 7.
+          For each group of type "LS" present in the offer, select the
+          local RtpTransceivers that are referenced by the MID values in the
+          specified group, and determine which of them
+          either reference a common local MediaStream
+          (specified in the calls to addTrack/addTransceiver used to create
+          them), or have no MediaStream to reference because they were not
+          created by addTrack/addTransceiver. If at least two such
+          RtpTransceivers exist, a
+          group of type "LS" with the mid values of these
+          RtpTransceivers MUST be added. Otherwise, this indicates a
```

Comment:
Isn't this same thing necessary for creating an offer? In other words, generate an LS group for all offered "m=" sections with no associated streams? This could happen either after:

`createOffer({offerToReceiveAudio:1, offerToReceiveVideo:1})`

Or an offer/answer in the reverse direction of the original:

```
pc1.addTrack(audio);
pc1.addTrack(video);
// Do offer/answer between pc1 and pc2
pc2.doSomethingThatRequiresNegotiation();
pc2.createOffer();
```